### ISSUE TITLE
Discord Embed not working 

### ISSUE THREAD

**Issue by sk3nn1xxx** [nextcloud/server#45898]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

I saw on GitHub that the feature already exists. The links sent to Discord should embed the MP4 or JPG directly in Discord, allowing the media to be viewed without having to click the link.

idk if i am too dumb to set up right but when i try to share an mp4 for example it looks like this:

![image](https://github.com/nextcloud/spreed/assets/62090337/59dad858-2f11-4fec-a9e6-4ffbdb328998)

or when i try to upload an image:

![image](https://github.com/nextcloud/spreed/assets/62090337/3a5ff4e7-9246-47db-9e9e-935208b77d8b)

**Comment by szaimen** [comment]
Hi, which nc version?

**Comment by sk3nn1xxx** [comment]
> Hi, which nc version?

29.0.2

**Comment by BlackWolfWoof** [comment]
use /preview for images and /download for videos. If you still have an issue with gifs for example use /download/filename.gif as example

**Comment by RuiRC** [comment]
> use /preview for images and /download for videos. If you still have an issue with gifs for example use /download/filename.gif as example

What does it mean to use /preview or /download ?

Needs more information

**Comment by BlackWolfWoof** [comment]
Its direct links to the files
You can do 4 things.
Link (Small preview in discord but not used to embed images properly. Use /preview)
Link/preview (image preview)
Link/download (direct link to file)
Link/download/image.png (direct link with file extension. Is needed for gifs on discord)

**Comment by RuiRC** [comment]
> Its direct links to the files You can do 4 things. Link (Small preview in discord but not used to embed images properly. Use /preview) Link/preview (image preview) Link/download (direct link to file) Link/download/image.png (direct link with file extension. Is needed for gifs on discord)

Thanks, after a bit more research I also figured that was the case. It's a bit of a shame nextcloud doesn't allow us to add these automatically to links as it would be a useful feature. Thank you for your reply though!

**Comment by BlackWolfWoof** [comment]
If you do Screenshots on desktop and use ShareX, this can be added
-------- Original Message --------On 25/10/2024 04:37, Rui  wrote:

Its direct links to the files You can do 4 things. Link (Small preview in discord but not used to embed images properly. Use /preview) Link/preview (image preview) Link/download (direct link to file) Link/download/image.png (direct link with file extension. Is needed for gifs on discord)

Thanks, after a bit more research I also figured that was the case. It's a bit of a shame nextcloud doesn't allow us to add these automatically to links as it would be a useful feature. Thank you for your reply though!

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/nextcloud/server/issues/45898#issuecomment-2436692876",
"url": "https://github.com/nextcloud/server/issues/45898#issuecomment-2436692876",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by RuiRC** [comment]
> If you do Screenshots on desktop and use ShareX, this can be added 

Yeah I'm aware of that myself, I've used ShareX for screenshots before. For my use case I'd really like to be able to just click the share button within Nextcloud and get a link already with what I want. I'd play around with creating an app or script for that but it's beyond what I can do.

**Comment by EntityinArray** [comment]
> Its direct links to the files You can do 4 things. Link (Small preview in discord but not used to embed images properly. Use /preview) Link/preview (image preview) Link/download (direct link to file) Link/download/image.png (direct link with file extension. Is needed for gifs on discord)

Doesn't work. I don't understand why it's impossible to just have a direct link to a file in Nextcloud `https://domain.com/files/my_video.mp4`. Like the way you can do with nginx or caddy
